### PR TITLE
fix: Add a build test to check all file names are in lower case

### LINKED ISSUE #18325: Add a build test to check all file names maintain case.
Unix does not differentiate between different file names, bases on lowercase/capitalcase

https://twitter.com/raisedadead/status/1050495401252581376

This is a cause of frustration, we need a CI test to handle this, and let it be known on the PR builds

### PR DESCRIPTION
**Pull request by sathish-kumar-subramani**
fixes issue #18325

- [x ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #18325

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -76,7 +76,7 @@
     "format:src": "prettier-eslint --write --trailing-comma none --single-quote './src/**/*.js'",
     "format:utils": "prettier-eslint --write --trailing-comma none --single-quote './utils/**/*.js'",
     "format": "npm run format:gatsby && npm run format:src && npm run format:utils",
-    "test": "jest"
+    "test": "sh check-file-name.sh && jest"
```
Comment (lynxlynxlynx) [inline client/package.json]: this should be bash explicitly, since you rely on its syntax.

File: client/check-file-name.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    filesToExclude=" -type f \\( ${filesToExclude} \\) -prune -o "
+fi
+
+command="find . ${dirsToExclude} ${filesToExclude} -name '*[A-Z]*' | wc -l"
+
+echo $command
+
+no="$(eval $command)"
+if (( $no > 0 )); then
+    echo "${no} file(s) are in upper case. Please rename files to lower case."
```
Comment (lynxlynxlynx) [inline client/check-file-name.sh]: For this to be more useful, rather save the find output, so you can present it here too.

**Review by lynxlynxlynx — CHANGES_REQUESTED** [review]

**Review by raisedadead — CHANGES_REQUESTED** [review]
Hi @Sathish-kumar-Subramani 

We would like to test this in JavaScript instead of a shell script. Are you able to do so? If not, please close this PR.

Thanks.

**Comment by Sathish-kumar-Subramani** [comment]
Hi @raisedadead 

Sure. I'll do with JS and update the pull request.
Can this wait until Saturday?

**Comment by raisedadead** [comment]
Hi @Sathish-kumar-Subramani 

Someone already added the tests. Thanks for taking the time to contribute. We look forward to futire contributions.

Thanks.

**Pull request closed without merging**